utils/build/docker/dotnet/parametric/Endpoints/ApmTestApi.cs

```diff
@@ -29,13 +29,20 @@ public static void MapApmTraceEndpoints(WebApplication app, ILogger<ApmTestApi>
         app.MapPost("/trace/span/flush", FlushSpans);
     }
 
+    private static readonly Assembly DatadogTraceAssembly = Assembly.Load("Datadog.Trace");
+    private const BindingFlags Instance = BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public;
+
     // Core types
     private static readonly Type SpanType = Type.GetType("Datadog.Trace.Span, Datadog.Trace", throwOnError: true)!;
     private static readonly Type SpanContextType = Type.GetType("Datadog.Trace.SpanContext, Datadog.Trace", throwOnError: true)!;
     private static readonly Type TracerType = Type.GetType("Datadog.Trace.Tracer, Datadog.Trace", throwOnError: true)!;
     private static readonly Type TracerManagerType = Type.GetType("Datadog.Trace.TracerManager, Datadog.Trace", throwOnError: true)!;
     private static readonly Type GlobalSettingsType = Type.GetType("Datadog.Trace.Configuration.GlobalSettings, Datadog.Trace", throwOnError: true)!;
-    private static readonly Type ImmutableTracerSettingsType = Type.GetType("Datadog.Trace.Configuration.ImmutableTracerSettings, Datadog.Trace", throwOnError: true)!;
+
+    // ImmutableTracerSettings was removed in 3.7.0
+    private static readonly Type TracerSettingsType = DatadogTraceAssembly.GetName().Version <= new Version(3, 6, 1, 0) ?
```

Comment:
just in case it helps, we used this as well:
https://github.com/DataDog/system-tests/blob/32e0b96a37ddd7a95f097f9e8a9da23578836a70/utils/build/docker/dotnet/weblog/Controllers/UsersController.cs#L1
defining the constant in the csproj
as it's set in `install_ddtrace.sh`